src/dbms/database.cpp

```diff
@@ -20,10 +22,15 @@ namespace memgraph::dbms {
 Database::Database(const storage::Config &config)
     : trigger_store_(config.durability.storage_directory / "triggers"),
       streams_{config.durability.storage_directory / "streams"} {
-  if (config.force_on_disk || utils::DirExists(config.disk.main_storage_directory)) {
+  auto storage_mode = memgraph::flags::ParseStorageMode();
+  if (storage_mode == memgraph::storage::StorageMode::ON_DISK_TRANSACTIONAL || config.force_on_disk ||
+      utils::DirExists(config.disk.main_storage_directory)) {
     storage_ = std::make_unique<storage::DiskStorage>(config);
   } else {
     storage_ = std::make_unique<storage::InMemoryStorage>(config);
+    if (storage_mode == memgraph::storage::StorageMode::IN_MEMORY_ANALYTICAL) {
+      storage_->SetStorageMode(storage_mode);
+    }
```

Comment:
What about changing the InMemoryStorage constructor to use `default_storage_mode` from config?

Reply:
hmm I don't understand: why would you change InMemoryStorage constructor

Reply:
`storage::InMemoryStorage` is constructed then, immediately the default storage mode that constructor uses wasn't good enough, hence you call `SetStorageMode`. Why not make it the constructor responsibility.